bindings/wasm/src/error.rs

```diff
@@ -110,3 +114,53 @@ impl From<identity::iota::BeeMessageError> for WasmError<'_> {
     }
   }
 }
+
+/// Convenience struct to convert Result<JsValue, JsValue> to an AccountResult<_, AccountError>
+pub struct JsValueResult(pub(crate) Result<JsValue>);
+
+impl From<Result<JsValue>> for JsValueResult {
+  fn from(result: Result<JsValue>) -> Self {
+    JsValueResult(result)
+  }
+}
+
+/// Implement From<JsValueResult> to AccountResult<_, AccountError> for each type
+macro_rules! impl_from_js_value {
+  ( $($t:ty),* ) => {
+    $(impl From<JsValueResult> for AccountResult<$t> {
+      fn from(result: JsValueResult) -> Self {
+        result
+          .0
+          .map_err(|js_value| AccountError::PromiseError(js_value.as_string().unwrap_or_default()))
+          .and_then(|js_value| {
+            js_value
+              .into_serde()
+              .map_err(|e| AccountError::InvalidJsValue(e.to_string()))
+          })
+      }
+    })*
+  };
+}
+
+impl_from_js_value!(
+  identity::account::DIDLease,
+  identity::crypto::PublicKey,
+  identity::account::Signature,
+  Option<identity::account::Generation>,
+  Option<identity::account::ChainState>,
+  Option<identity::account::IdentityState>,
+  bool
+);
```

Comment:
Hmm, it's really annoying that we have to deserialize through `JsValue` as a result of `js_sys::Promise`. I don't see a way around this though...